### PR TITLE
[Merged by Bors] - feat: port Data.Holor

### PR DESCRIPTION
**Pull request by qawbecrdtey**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Holor.lean [hunk 1]
```diff
@@ -0,0 +1,431 @@
...
+/-- Remove elements which correspond to `ds₁`. -/
+def take : ∀ {ds₁ : List ℕ}, HolorIndex (ds₁ ++ ds₂) → HolorIndex ds₁
+  | ds, is => ⟨List.take (length ds) is.1, forall₂_take_append is.1 ds ds₂ is.2⟩
+#align holor_index.take HolorIndex.take
+
+-- Porting note: added this doc
+/-- Remove elements which correspond to `ds₂`. -/
+def drop : ∀ {ds₁ : List ℕ}, HolorIndex (ds₁ ++ ds₂) → HolorIndex ds₂
+  | ds, is => ⟨List.drop (length ds) is.1, forall₂_drop_append is.1 ds ds₂ is.2⟩
+#align holor_index.drop HolorIndex.drop
```
Comment (Komyyy) [inline Mathlib/Data/Holor.lean:66]: My English is not so good, please correct these new docs.

File: Mathlib/Data/Holor.lean [hunk 2]
```diff
@@ -0,0 +1,431 @@
...
+/-- Cast a index of a holor by associativity of the list. -/
+def assocRight : HolorIndex (ds₁ ++ ds₂ ++ ds₃) → HolorIndex (ds₁ ++ (ds₂ ++ ds₃)) :=
+  cast (congr_arg HolorIndex (append_assoc ds₁ ds₂ ds₃))
+#align holor_index.assoc_right HolorIndex.assocRight
+
+-- Porting note: added this doc
+/-- Cast a index of a holor by associativity of the list. -/
+def assocLeft : HolorIndex (ds₁ ++ (ds₂ ++ ds₃)) → HolorIndex (ds₁ ++ ds₂ ++ ds₃) :=
+  cast (congr_arg HolorIndex (append_assoc ds₁ ds₂ ds₃).symm)
+#align holor_index.assoc_left HolorIndex.assocLeft
```
Comment (Komyyy) [inline Mathlib/Data/Holor.lean:82]: My English is not so good, please correct these new docs.

File: Mathlib/Data/Holor.lean [hunk 3]
```diff
@@ -0,0 +1,431 @@
...
+/-- Cast a holor by associativity of the list. -/
+def assocRight : Holor α (ds₁ ++ ds₂ ++ ds₃) → Holor α (ds₁ ++ (ds₂ ++ ds₃)) :=
+  cast (congr_arg (Holor α) (append_assoc ds₁ ds₂ ds₃))
+#align holor.assoc_right Holor.assocRight
+
+-- Porting note: added this doc
+/-- Cast a holor by associativity of the list. -/
+def assocLeft : Holor α (ds₁ ++ (ds₂ ++ ds₃)) → Holor α (ds₁ ++ ds₂ ++ ds₃) :=
+  cast (congr_arg (Holor α) (append_assoc ds₁ ds₂ ds₃).symm)
+#align holor.assoc_left Holor.assocLeft
```
Comment (Komyyy) [inline Mathlib/Data/Holor.lean:210]: My English is not so good, please correct these new docs.

**Comment by jcommelin** [comment]
@Komyyy We've decided that adding docs is orthogonal to the porting process, and so we've disabled the docBlame linter. Feel free to just remove these docs again. Then some expert on holors (not me!) can add them later.

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**